### DISCUSSION TITLE
How to generate shared libraries

### DISCUSSION THREAD

**Discussion by MewanCdot** [strongswan/strongswan#2470] [General]
I have gone through the documentation and I did not find instructions on how to generate the shared libraries. I am following this tutorial (https://github.com/X-dea/flutter_vpn) to build a vpn flutter app that uses strongswan for the android side and it uses libstrongswan.so, libipsec.so, libimcv.so, etc. The reason i want to generate my own shared libraries is because the current application i am working on is using strongswan-v6.0.0 and this tutorial im following only has v5.9.5

**Comment by tobiasbrunner** [comment]
I can only point you to the [Android](https://docs.strongswan.org/docs/5.9/os/android.html) and [Android app](https://docs.strongswan.org/docs/5.9/os/androidVpnClientBuild.html) build instructions.
